tests/test_Sequence.cc

```diff
@@ -3566,3 +3567,217 @@ TEST_CASE("Sequence: terminate sequence with Lua exit function", "[Sequence]")
+
+    Step step_action_3{Step::type_action};
+    step_action_3.set_script("d = preface('Eve!')");
+    step_action_3.set_used_context_variable_names({VariableName{"d"}});
+
+    Sequence seq{ "test_sequence" };
+    seq.push_back(step_action_1);
+    seq.push_back(step_action_2);
+    seq.push_back(step_action_3);
+
+    seq.execute(ctx, nullptr);
+
+    REQUIRE(std::get<VarString>(ctx.variables["a"]) == "Alice calls Bob!");
+    REQUIRE(std::get<VarString>(ctx.variables["b"]) == "Alice calls Charlie!");
+    REQUIRE(std::get<VarString>(ctx.variables["c"]) == "Bob calls Marvin!");
+    REQUIRE(std::get<VarString>(ctx.variables["d"]) == "Alice calls Eve!");
+}
+
+TEST_CASE("Sequence: Check line number on failure (setup at line 2)", "[Sequence]")
+{
+    Context ctx;
+    ctx.step_setup =
+        R"(preface = 'Alice'
+           this line will fail
+        )";
+
+    Step step_action_1{Step::type_action};
+    step_action_1.set_script("a = preface .. ' and Bob'");
+    step_action_1.set_used_context_variable_names({VariableName{"a"}});
+
+    Sequence seq{ "test_sequence" };
+    seq.push_back(step_action_1);
+
+    try
+    {
+        seq.execute(ctx, nullptr);
+    }
+    catch(const Error& e)
+    {
+        REQUIRE_THAT(e.what(), Catch::Matchers::Contains("error: [setup] 2"));
+    }
+    catch(...)
+    {
+        FAIL("Must throw Error exception");
+    }
+}
+
+TEST_CASE("Sequence: Check line number on failure (script at line 2)", "[Sequence]")
+{
+    Context ctx;
+    ctx.step_setup = "preface = 'Alice'";
+
+    Step step_action_1{Step::type_action};
+    step_action_1.set_script(
+        R"(a = preface .. ' and Bob'
+           Oops this line will fail
+        )");
+    step_action_1.set_used_context_variable_names({VariableName{"a"}});
+
+    Sequence seq{ "test_sequence" };
+    seq.push_back(step_action_1);
+
+    try
+    {
+        seq.execute(ctx, nullptr);
+    }
+    catch(const Error& e)
+    {
+        REQUIRE_THAT(e.what(), Catch::Matchers::Contains("error: 2"));
+    }
+    catch(...)
+    {
+        FAIL("Must throw Error exception");
+    }
+}
+
+TEST_CASE("Sequence: Check line number on failure (script at line 3)", "[Sequence]")
+{
+    Context ctx;
+    ctx.step_setup = "preface = 'Alice'";
+
+    Step step_action_1{Step::type_action};
+    step_action_1.set_script(
+        R"(a = preface .. ' and Bob'
+           a = 1
+           And again this line will fail
+        )");
+    step_action_1.set_used_context_variable_names({VariableName{"a"}});
+
+    Sequence seq{ "test_sequence" };
+    seq.push_back(step_action_1);
+
+    try
+    {
+        seq.execute(ctx, nullptr);
+    }
+    catch(const Error& e)
+    {
+        REQUIRE_THAT(e.what(), Catch::Matchers::Contains("error: 3"));
+    }
+    catch(...)
+    {
+        FAIL("Must throw Error exception");
+    }
+}
+
+TEST_CASE("Sequence: Change setup script during execution", "[Sequence]")
```

Comment:
Regarding question from [Lars](https://github.com/taskolib/taskolib/pull/49#pullrequestreview-1202103233).

Since I moved the setup script as data to `Context` we don't need any setter/getter member functions in `Sequence`. But is this the expected behaviour because now a user can manipulate the setup script on the fly during execution the sequence. I added this test as proof.

This must be clarified somehow. If we decide to move it to `Sequence` then I will also expand the serialisation functionality. For `Context` and most likely `Context::step_setup` we don't have any serialisation.

Reply:
Maybe one solution to still keep `Context::step_setup` is to make a copy of `Context::step_setup` in `Sequence::execute(Context&)` and have a getter function `Sequence::get_step_setup():const std::string&`... 🤔

Reply:
I think what we agreed upon was this:
- `Sequence` stores a member `step_setup_script_` with a getter and a setter.
- Sequence serialization and deserialization store and restore this step setup script.
- `Context` also has a member `step_setup_script`, but it is documented that its contents are overwritten whenever a sequence is started.
- Within `Sequence::execute()`, the `step_setup_script_` is copied into `Context::step_setup_script`, just before the sequence is started.

This way the step setup script is a part of the sequence, but parameter passing happens via the context.

Reply:
I added setter/getter member functions for `Sequence` and also improved the docu. The serialisation was a little bit complex but I managed it (except one question - see the other request from me).